pkg/network/encoding/json.go

```diff
@@ -23,8 +23,9 @@ type jsonSerializer struct {
 }
 
 func (j jsonSerializer) Marshal(conns *network.Connections, writer io.Writer) error {
+	connectionsModeler := NewConnectionsModeler(conns)
```

Comment:
don't you need to close the modeler?

Reply:
the connection modeler should be created once and not everytime we call marshal (preparation for batching)